### ISSUE TITLE
Spray is not OSGi-compatible

### ISSUE THREAD

**Issue by FeiWongReed** [spray/spray#1067]
Spray itself has OSGi manifests, but they points to non-OSGI-enabled libraries and, actually, incorrect.

Here they are:

org.parboiled:*
com.chuusai:shapeless

I resolved problem for parboiled (https://github.com/sirthias/parboiled/pull/92), and I hope it will be accepted.

I'm also going to do the same for shapeless and other your dependencies (if any). But could you help me to enforce OSGification of your dependencies?

**Comment by FeiWongReed** [comment]
Actually, it's possible to run Spray under OSGi environment even for now.

You should manually compile shapeless v1 and parboiled v1 with my patches:
https://github.com/sirthias/parboiled/pull/92
https://github.com/milessabin/shapeless/pull/437

You should use ;clean;packagedArtifacts SBT command to do that.

Then you should install these artifacts into your container manually.
Then you may install Spray and your application and everything should work.

**Comment by jrudolph** [comment]
Yes, we don't have the capacity and expertise to maintain OSGI support. We will accept PRs to fix it as long as they don't break anything else (which unfortunately has happened in the past with OSGI support).

Thanks for providing the fixes. I'm leaving this open for now for reference.
